:PROPERTIES:
:ID:       8adf45de-c02e-427e-b8bd-ad0b169a5d34
:END:
#+TITLE: Bioinformatics
#+DESCRIPTION: 
#+TAGS:

* Docs

* Resources

* Topics
** Data Sources
*** Overviews
+ [[https://www.ncbi.nlm.nih.gov/guide/all/][NIH/NLM: All NCBI Resources]]

*** Sparqling Genomics

Putting this here since it contains a lot of discussion of data sources.

+ The [[https://gitlab.com/roelj/sparqling-genomics/-/blob/main/doc/sparqling-genomics.pdf?ref_type=heads][PDF]] docs from [[https://gitlab.com/roelj/sparqling-genomics][roelj/sparqling-genomics]] are a fantastic introduction to the
  data sources, especially ontological sources (see [[https://www.roelj.com/sparqling-genomics.html][blog]], [[https://orcid.org/0000-0003-4324-5350][orcid]] and [[https://www.youtube.com/watch?v=guzkA43wAYY][video]]...).
  It's built with Guile scheme using [[https://github.com/roelj/guile-sparql][roelj/guile-sparql]].

+ (2024) [[http://dx.doi.org/10.48550/arXiv.2407.20879][A Scalable Tool For Analyzing Genomic Variants Of Humans Using
  Knowledge Graphs and Machine Learning]]
+ (2018) [[https://aisberg.unibg.it/handle/10446/134460?mode=complete][SPARQLING: painlessly drawing SPARQL queries over GRAPHOL ontologies]]
  ([[https://github.com/picorana/painless_sparql][picorana/painless_sparql]]) which is built on [[https://github.com/obdasystems/sparqling][obdasystems/sparqling]]

And if you need to run this analysis in a DPDK data plane (~100 Gb/s lmao), then
there's this analysis: (2024) [[https://doi.org/10.1109/CLOUD62652.2024.00044][A Technique for Secure Variant Calling on Human
Genome Sequences Using SmartNICs]] ... hippies can use the link in the
[[https://github.com/MU-Data-Science/GAF][MU-Data-Science/GAF]] github (otherwise you're SOL bc it's IEEE)

*** Aggregators
**** [[https://www.wikidata.org][wikidata.org]]

**** [[wikipedia:List_of_biological_databases]]

These are just the links to the Wiki. I'm hoping to extract more information
about the databases without needing to click on each page or script the
extraction of each URL

| Sequence Databases          | Secondary Database           | Other Databases                 | Other Specialized Databases          |
|-----------------------------+------------------------------+---------------------------------+--------------------------------------|
| [[GenBank]]                     | [[wikipedia:UniProt][UniProt]]                      | [[wikipedia:BioNumbers][BioNumbers]]                      | [[Barcode of Life Data System]]          |
| [[European Nucleotide Archive]] | [[wikipedia:UniProt#UniProtKB/Swiss-Prot][UniProt#UniProtKB/Swiss-Prot]] | [[wikipedia:Ensembl_genome_database_project][Ensembl Genome Database Project]] | [[Saccharomyces Genome Database]]        |
| [[DNA Data Bank of Japan]]      | [[wikipedia:UniProt#UniProtKB/TrEMBL][UniProt#UniProtKB/TrEMBL]]     | [[wikipedia:InterPro][InterPro]]                        | [[FlyBase]]                              |
| [[China National GeneBank]]     | [[wikipedia:Protein_Information_Resource][Protein Information Resource]] | [[wikipedia:KEGG][KEGG]]                            | [[VectorBase]]                           |
|                             |                              | [[wikipedia:Gene_Ontology][Gene Ontology]]                   | [[WormBase]]                             |
|                             |                              |                                 | [[Rat Genome Database]]                  |
|                             |                              |                                 | [[PHI-base]]                             |
|                             |                              |                                 | [[The Arabidopsis Information Resource]] |
|                             |                              |                                 | [[GISAID]]                               |
|                             |                              |                                 | [[Zebrafish Information Network]]        |
|-----------------------------+------------------------------+---------------------------------+--------------------------------------|

*** NCBI

From the "Database" tab of the "All Resources" page on NCBI

#+begin_example javascript
db = Array.from(document.querySelectorAll("#databases > dl > dt > a"))
db.map((a) => `| [[${a.href}][${a.text}]] | |`).join("\n")
#+end_example

And manually add the column headers bc I'm lazy

| Link                                                 | Notes |
|------------------------------------------------------+-------|
| [[https://www.ncbi.nlm.nih.gov/assembly][Assembly]]                                             |       |
| [[https://www.ncbi.nlm.nih.gov/biocollections][BioCollections]]                                       |       |
| [[https://www.ncbi.nlm.nih.gov/bioproject][BioProject (formerly Genome Project)]]                 |       |
| [[https://www.ncbi.nlm.nih.gov/biosample][BioSample]]                                            |       |
| [[https://www.ncbi.nlm.nih.gov/books][Bookshelf]]                                            |       |
| [[https://www.ncbi.nlm.nih.gov/clinvar][ClinVar]]                                              |       |
| [[https://clinicaltrials.gov/][ClinicalTrials.gov]]                                   |       |
| [[https://www.ncbi.nlm.nih.gov/Structure/][Computational Resources from NCBI's Structure Group ]] |       |
| [[https://www.ncbi.nlm.nih.gov/projects/CCDS/CcdsBrowse.cgi][Consensus CDS (CCDS)]]                                 |       |
| [[https://www.ncbi.nlm.nih.gov/cdd][Conserved Domain Database (CDD)]]                      |       |
| [[https://www.ncbi.nlm.nih.gov/dbvar][Database of Genomic Structural Variation (dbVar)]]     |       |
| [[https://www.ncbi.nlm.nih.gov/gap][Database of Genotypes and Phenotypes (dbGaP)]]         |       |
| [[https://www.ncbi.nlm.nih.gov/snp][Database of Short Genetic Variations (dbSNP)]]         |       |
| [[https://www.ncbi.nlm.nih.gov/genbank/][GenBank]]                                              |       |
| [[https://www.ncbi.nlm.nih.gov/gene][Gene]]                                                 |       |
| [[https://www.ncbi.nlm.nih.gov/geo/][Gene Expression Omnibus (GEO) Database ]]              |       |
| [[https://www.ncbi.nlm.nih.gov/gds][Gene Expression Omnibus (GEO) Datasets]]               |       |
| [[https://www.ncbi.nlm.nih.gov/geoprofiles/][Gene Expression Omnibus (GEO) Profiles]]               |       |
| [[https://www.ncbi.nlm.nih.gov/books/NBK1116/][GeneReviews]]                                          |       |
| [[https://www.ncbi.nlm.nih.gov/books/NBK22183/][Genes and Disease]]                                    |       |
| [[https://www.ncbi.nlm.nih.gov/gtr/][Genetic Testing Registry (GTR)]]                       |       |
| [[https://www.ncbi.nlm.nih.gov/genome][Genome]]                                               |       |
| [[https://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/][Genome Reference Consortium (GRC)]]                    |       |
| [[https://www.ncbi.nlm.nih.gov/glycans/index.html][Glycans]]                                              |       |
| [[https://www.ncbi.nlm.nih.gov/RefSeq/HIVInteractions][HIV-1, Human Protein Interaction Database]]            |       |
| [[https://www.ncbi.nlm.nih.gov/ipg][Identical Protein Groups]]                             |       |
| [[https://www.ncbi.nlm.nih.gov/genomes/FLU/FLU.html][Influenza Virus]]                                      |       |
| [[https://www.ncbi.nlm.nih.gov/nlmcatalog/journals][Journals in NCBI Databases]]                           |       |
| [[https://www.ncbi.nlm.nih.gov/mesh][MeSH Database]]                                        |       |
| [[https://www.ncbi.nlm.nih.gov/medgen][MedGen]]                                               |       |
| [[https://ncbi.github.io/cxx-toolkit/][NCBI C++ Toolkit Manual]]                              |       |
| [[https://www.ncbi.nlm.nih.gov/education/][NCBI Education Page]]                                  |       |
| [[https://www.ncbi.nlm.nih.gov/books/NBK21106/][NCBI Glossary]]                                        |       |
| [[https://www.ncbi.nlm.nih.gov/books/NBK143764/][NCBI Handbook]]                                        |       |
| [[https://www.ncbi.nlm.nih.gov/books/NBK3831/][NCBI Help Manual]]                                     |       |
| [[https://www.ncbi.nlm.nih.gov/pathogens/][NCBI Pathogen Detection Project]]                      |       |
| [[https://www.ncbi.nlm.nih.gov/nlmcatalog][National Library of Medicine (NLM) Catalog]]           |       |
| [[https://www.ncbi.nlm.nih.gov/nuccore][Nucleotide Database]]                                  |       |
| [[https://www.ncbi.nlm.nih.gov/omim][Online Mendelian Inheritance in Man (OMIM)]]           |       |
| [[https://www.ncbi.nlm.nih.gov/popset][PopSet]]                                               |       |
| [[https://www.ncbi.nlm.nih.gov/proteinclusters][Protein Clusters]]                                     |       |
| [[https://www.ncbi.nlm.nih.gov/protein][Protein Database]]                                     |       |
| [[https://www.ncbi.nlm.nih.gov/protfam][Protein Family Models]]                                |       |
| [[https://www.ncbi.nlm.nih.gov/pcassay][PubChem BioAssay]]                                     |       |
| [[https://www.ncbi.nlm.nih.gov/pccompound][PubChem Compound]]                                     |       |
| [[https://www.ncbi.nlm.nih.gov/pcsubstance][PubChem Substance]]                                    |       |
| [[https://www.ncbi.nlm.nih.gov/pubmed][PubMed]]                                               |       |
| [[https://www.ncbi.nlm.nih.gov/pmc/][PubMed Central (PMC)]]                                 |       |
| [[https://www.ncbi.nlm.nih.gov/refseq/rsg/][RefSeqGene]]                                           |       |
| [[https://www.ncbi.nlm.nih.gov/RefSeq/][Reference Sequence (RefSeq)]]                          |       |
| [[https://www.ncbi.nlm.nih.gov/retroviruses/][Retrovirus Resources]]                                 |       |
| [[https://www.ncbi.nlm.nih.gov/genomes/SARS/SARS.html][SARS CoV]]                                             |       |
| [[https://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?][Sequence Read Archive (SRA)]]                          |       |
| [[https://www.ncbi.nlm.nih.gov/sites/entrez?db=structure][Structure (Molecular Modeling Database)]]              |       |
| [[https://www.ncbi.nlm.nih.gov/taxonomy][Taxonomy]]                                             |       |
| [[https://www.ncbi.nlm.nih.gov/genbank/TPA.html][Third Party Annotation (TPA) Database]]                |       |
| [[https://www.ncbi.nlm.nih.gov/genomes/GenomesHome.cgi?][Viral Genomes]]                                        |       |
| [[https://www.ncbi.nlm.nih.gov/genomes/VirusVariation/][Virus Variation]]                                      |       |

*** Natural Products

**** [[https://lotus.naturalproducts.net/][lotus.naturalproducts.net]]

LOTUS documents the occurance of ingredients in natural products

**** Ontological

GO: Gene Ontology

** Graphical notation
*** Process Diagram

[[https://www.celldesigner.org/documents/ProcessDiagram.html][The Process Diagram: Rationale and Definition]]

[[file:img/med/cell-designer-notation.png]]

*** SBGN [[https://sbgn.github.io/learning][Systems Biological graphical notation]]

[[file:img/med/bio-process-diagram-notation.png]]


* Ideas

** Protein Folding

Energy is a statistical concept -- connect Boltzmann's ideas on entropy
generally to configuration space & entropy and to the Free Energy Landscape.
With either classical statistical mechanics or quantum, the configurations of
molecules some unit volume of space are more likely when there is less free
energy. You can calculate by integrating over the Laplacian. For some unit of
volume frozen in time, where the laplacian is high, the forces between molecules
are also high.

This concept of energy as statistics always resulted in novelty w.r.t /how I
personally understood protein folding or biophysics./ No one ever took me
seriously, but I consistently found that any novel thought I had on protein
folding was 1-5 years behind current research on computational folding. I'm sure
this happens all the time for anyone who is smart, creative and alone.

*** Paradox of Consistent Protein Structure

We know proteins form consistently despite flucations in temperature or pH or
other confounding factors ... yet, before AlphaFold, protein folding and
structure were traditionally very difficult to resolve.

I can't remember when I first considered this, but when AlphaFold was released
(December 2018), the results confirmed what I suspected which is what

**** TODO describe

*** Ribosomes partially folded protein subunits and amino acid orientation

How essential is tRNA specificity to both codon affinity and amino affinity?

I've thought a lot about this and I think the reduction of bioinformatics
sequences to strings of animos confounds the folding problem. Anyways, some of
this is new.

+ The partially folded protein should block motion of tRNAs which, when bound to
  an amino, should release potential energy when depositing it onto the
  incomplete protein.
+ This partially blocked motion should bias the local distribution of tRNAs.
+ The tRNAs bound to aminos are somewhat heavy and somewhat bulky, so the tRNA
  anticodon competition over access to exposed mRNA codons should pose many
  mechanical/hydrostatic constraints. This should be found as a rate-limiting
  constrain on ribosomal

The weak bonds that contribute generally to tRNAs "valance" for amino acids
should be focused in their orientation (biased towards one side or one
location).

**** Symmetry, Groups and Life

This tendency to create bias towards one side or another is typical of life.
Form is essential to function and form is not arbitrary.
mamallian gestation, life creates differentiated form by accumulating
constraints on symmetries. See below.

+ At various scales in, life biases the the orientation of
  forces/charges/structures to eliminate radial/mirror symmetries. The "human
  form" is not entirely encoded into DNA: our programming relies on
  environmental pressures/influences that promote its convergence towards
  various "platonic form." To avoid the Woo Woo Genetics, you might instead say
  that life differentiates its macroscopic for on a path through increasingly
  specific Euclidean Groups (like [[https://en.wikipedia.org/wiki/Point_groups_in_three_dimensions][point groups]] or linear groups whose
  constraints can be specified with [[https://en.wikipedia.org/wiki/Coxeter_notation][Coxeter notation]])
+ A fetus forms a spherical blastoid, which has 2 degrees of radial symmetry and
  3 degrees of mirror symmetry. Whether along an electromagnetic potential or
  biophysical potential, it develops an inner polarization that informs the
  biophysical processes to prefer orientation along an axis.
+ The Gastrula forms into a torus (or toroid) along this basis, placing a 180°
  constraint on one axis of rotational symmetry. This forms another basis along
  which a biophysical potential can emerge -- i.e. the torus now properly has an
  "up/down" or "positive negative" orientation.
+ The positive/negative ends of the torus then have a basis upon which to
  upregulate/downregulate genetic programs. The fetus eventually begins to form
  a spine. As it does, the original difference in positive/negative causes the
  fetus to form a head and an ass.
+ The original 180° rotational symmetry is now a 360° rotational symmetry. The
  differentiation between the upper/lower spine also breaks the vertical mirror
  symmetry.

Sound like Terrance McKenna Woo Woo Genetics? See Harvard Lecture [[https://www.youtube.com/watch?v=hc2K0ZdkI8A&list=PL0NRmB0fnLJT6I0RExpxUhS1T8rGPEdaE&index=1&t=150s][The Shape of
Things to Come]] on morphgenetics (McKenna wrote about a morphogenetic field,
which)

This kinda stuff emerges throughout biology, even in the biomechanics of
cells. The organelles/proteins are just pieces that fit into a comprehensive
system. Evolution is a random process, but selection will not prefer arbitrary
parameters/phenotypes for long. Selection/reproduction will tend to reward
resonant dynamics & mechanics controled by postive/negative feedback loops which
can be constrained.

***** TODO finish this & next

*** Prioritize Protein Structures with Unlikely Potential Energy



+ Ligands typically have small molar masses ranging from 150 for small ligands up
  to 1500+ for larger ligands.
+ Proteins have extremely large molar masses and their constituent monomers (the
  amino acids) have restricted degrees of freedom.
+ For aminos or protein components like beta sheets or alpha helices, the
  constraints on DoF funnel the mechanical response of the protein along the
  surface of a very small region of free energy landscape (FES) -- esp. when
  compared to hypothetical FES without those constraints.
+ The constraints are interdependent w.r.t. the monomers, protein components and
  larger structure. The dependency graph is undirected and can contains
  loops. The loops can create positive feedback that strengthens those
  constraints' dynamics.
+ Energy can be seen as a statistical concept. High potential energy is
  statistically unlikely without more information. With proper positive or
  negative feedback mechanisms, the constraints on protein components can shape
  the protein's assembly through its folding funnel -- i.e. some paths down the
  folding funnel are more likely than they may otherwise appear to be.
+


   where the
amino acids' DoF have interdependent constraints)

they trigger a cascade of
downstream reactions

* Issues
